lib/metrics/AmqpMetricReporter.ts

```diff
@@ -22,10 +22,23 @@ import {
   ScheduledMetricReporterOptions,
   Scheduler,
   StdClock,
+  Tags,
   Timer,
   TimeUnit,
 } from "inspector-metrics";
 
+/**
+ * Interface for building a message for a metric.
+ */
+export type MetricMessageBuilder = (registry: MetricRegistry, metric: Metric, type: MetricType, date: Date, tags: Tags) => Amqp.Message;
```

Comment:
Since we return null in the default implementation the return type should be
```suggestion
export type MetricMessageBuilder = (registry: MetricRegistry, metric: Metric, type: MetricType, date: Date, tags: Tags) => Amqp.Message | null;
```